rs/registry/canister/src/mutations/node_management/do_remove_node_directly.rs

```diff
@@ -35,12 +35,29 @@ impl Registry {
             })
             .unwrap();
 
-        // 2. Get the caller ID and check that it matches the node's NO
-        assert_eq!(
-            node_operator_id, caller_id,
-            "The caller {}, does not match this Node's Operator id.",
-            caller_id
-        );
+        // 2. Compare the caller_id (node operator) with the node's node operator and, if that fails,
+        // fall back to comparing the node provider ID of the caller and the node's node provider ID.
+        if caller_id != node_operator_id {
+            let node_provider_caller = get_node_provider_id_for_operator_id(self, caller_id)
+                .map_err(|e| {
+                    format!(
+                        "{}do_remove_node_directly: Aborting node removal: {}",
+                        LOG_PREFIX, e
+                    )
+                });
+            let node_provider_of_the_node =
+                get_node_provider_id_for_operator_id(self, node_operator_id).map_err(|e| {
+                    format!(
+                        "{}do_remove_node_directly: Aborting node removal: {}",
+                        LOG_PREFIX, e
+                    )
+                });
+            assert_eq!(
+                node_provider_caller, node_provider_of_the_node,
```

Comment:
Wouldn't this mean that _any_ node_operator could deregister or modify any node of the node operator?  

Doesn't that make the loss of a single node operator key much more dangerous?